api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CXCASCRNCohortQueries.java

```diff
@@ -77,6 +78,392 @@ private CohortDefinition getA() {
+      CXCASCRNCohortQueries.CXCASCRNResult cxcascrnResult, boolean beforeStartDate) {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName("Patients with Screening Result for Cervical Cancer");
+    cd.addParameter(new Parameter("startDate", "startDate", Date.class));
+    cd.addParameter(new Parameter("endDate", "endDate", Date.class));
+    cd.addParameter(new Parameter("location", "location", Location.class));
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("28", hivMetadata.getRastreioDoCancroDoColoUterinoEncounterType().getEncounterTypeId());
+    map.put("2094", hivMetadata.getResultadoViaConcept().getConceptId());
+    map.put("664", hivMetadata.getNegative().getConceptId());
+    map.put("703", hivMetadata.getPositive().getConceptId());
+    map.put("2093", hivMetadata.getSuspectedCancerConcept().getConceptId());
+    map.put("165436", hivMetadata.getHumanPapillomavirusDna().getConceptId());
+
+    StringBuilder query =
+        new StringBuilder(
+            "SELECT final.patient_id "
+                + "FROM   ( "
+                + "           SELECT   via_or_hpv.patient_id, "
+                + "                    Max(via_or_hpv.screening_date) AS last_screening_date "
+                + "           FROM     ( "
+                + "                        SELECT     p.patient_id, "
+                + "                                   Date(e.encounter_datetime) AS screening_date "
+                + "                        FROM       patient p "
+                + "                                       INNER JOIN encounter e "
+                + "                                                  ON         e.patient_id = p.patient_id "
+                + "                                       INNER JOIN obs o "
+                + "                                                  ON         o.encounter_id = e.encounter_id "
+                + "                        WHERE      p.voided = 0 "
+                + "                          AND        e.voided = 0 "
+                + "                          AND        o.voided = 0 "
+                + "                          AND        e.encounter_type = ${28} "
+                + "                          AND        o.concept_id = ${2094} ");
+    if (cxcascrnResult == CXCASCRNCohortQueries.CXCASCRNResult.NEGATIVE) {
+      query.append("    AND o.value_coded = ${664} ");
+    } else if (cxcascrnResult == CXCASCRNCohortQueries.CXCASCRNResult.POSITIVE) {
+      query.append("    AND o.value_coded = ${703} ");
+    } else if (cxcascrnResult == CXCASCRNCohortQueries.CXCASCRNResult.SUSPECTED) {
+      query.append("    AND o.value_coded = ${2093} ");
+    } else if (cxcascrnResult == CXCASCRNCohortQueries.CXCASCRNResult.ANY) {
+      query.append("    AND o.value_coded IN (${2093}, ${664}, ${703}) ");
+    }
+    query.append("                          AND        e.encounter_datetime  ");
```

Comment:
If you are using DATE function on the select clause, it should be used on `e.encounter_datetime` verification's either.

Reply:
Done